dep0/lib/05_llvmgen/src/gen_alloca.cpp

```diff
@@ -0,0 +1,48 @@
+#include "private/gen_alloca.hpp"
+
+#include "private/gen_type.hpp"
+#include "private/gen_val.hpp"
+
+#include "dep0/match.hpp"
+
+namespace dep0::llvmgen {
+
+needs_alloca_result_t needs_alloca(typecheck::expr_t const& type)
+{
+    if (auto const app = get_if_app_of_array(type))
+        return needs_alloca_result::array_t{app->args[0ul], app->args[1ul]};
```

Comment:
shouldn't this pass an `array_properties_t` instead? In fact `gen_alloca_if_needed()` seems to be allocating a vector of pointers if this was a multidimensional array

Reply:
indeed; fixed